server/src/main/java/org/elasticsearch/monitor/jvm/HotThreads.java

```diff
@@ -104,6 +105,33 @@ public static void logLocalHotThreads(Logger logger, Level level, String prefix,
         }
     }
 
+    /**
+     * Capture and log the current threads on the local node. Unlike hot threads this does not sample and captures current state only.
+     * Useful for capturing stack traces for unexpectedly-slow operations in production. The resulting message might be large, so it is
+     * split per thread and logged as multiple entries.
+     *
+     * @param logger        The logger to use for the logging
+     * @param level         The log level to use for the logging.
+     * @param prefix        The prefix to emit on each chunk of the logging.
+     */
+    public static void logLocalCurrentThreads(Logger logger, Level level, String prefix) {
```

Comment:
what was the specific use case that would benefit from this? Maybe we could use it there with this PR too? otherwise this seems unused.

Reply:
Correct, it is currently unused. The goal of the pr is to introduce the mechanism to log hot threads in plain threads without changing existing behavior. Existing `logLocalHotThreads` usages could be moved to `logLocalCurrentThreads` eventually. Meanwhile this change could help with serverless performance investigations.